src/reanimated2/layoutReanimation/web/domUtils.ts

```diff
@@ -1,63 +1,114 @@
+
+// Since we cannot remove keyframe from DOM by its name, we have to store its id
+const animationNameToIndex = new Map<string, number>();
+const animationNameList: string[] = [];
 
 /**
  *  Creates `HTMLStyleElement`, inserts it into DOM and then inserts CSS rules into the stylesheet.
  *  If style element already exists, nothing happens.
  */
 export function configureWebLayoutAnimations() {
-  if (document.getElementById(WEB_ANIMATIONS_ID) !== null) {
+  if (
+    !isWindowAvailable() ||
+    document.getElementById(PREDEFINED_WEB_ANIMATIONS_ID) !== null
+  ) {
     return;
   }
 
-  const style = document.createElement('style');
-  style.id = WEB_ANIMATIONS_ID;
+  const predefinedAnimationsStyleTag = document.createElement('style');
+  predefinedAnimationsStyleTag.id = PREDEFINED_WEB_ANIMATIONS_ID;
 
-  style.onload = () => {
-    if (!style.sheet) {
+  predefinedAnimationsStyleTag.onload = () => {
+    if (!predefinedAnimationsStyleTag.sheet) {
       console.error(
-        '[Reanimated] Failed to create layout animations stylesheet'
+        '[Reanimated] Failed to create layout animations stylesheet.'
       );
       return;
     }
 
     for (const animationName in Animations) {
-      style.sheet.insertRule(Animations[animationName as AnimationNames].style);
+      predefinedAnimationsStyleTag.sheet.insertRule(
+        Animations[animationName as AnimationNames].style
+      );
     }
   };
 
-  document.head.appendChild(style);
+  const customAnimationsStyleTag = document.createElement('style');
+  customAnimationsStyleTag.id = CUSTOM_WEB_ANIMATIONS_ID;
+
+  document.head.appendChild(predefinedAnimationsStyleTag);
+  document.head.appendChild(customAnimationsStyleTag);
 }
 
 export function insertWebAnimation(animationName: string, keyframe: string) {
+  if (!isWindowAvailable()) {
+    return;
+  }
+
   const styleTag = document.getElementById(
-    WEB_ANIMATIONS_ID
+    CUSTOM_WEB_ANIMATIONS_ID
   ) as HTMLStyleElement;
 
   if (!styleTag.sheet) {
-    console.error('[Reanimated] Failed to create layout animations stylesheet');
+    console.error(
+      '[Reanimated] Failed to create layout animations stylesheet.'
+    );
     return;
   }
 
-  const customTransitionId = styleTag.sheet.insertRule(keyframe);
-  customAnimations.set(animationName, customTransitionId);
+  styleTag.sheet.insertRule(keyframe, 0);
+  animationNameList.unshift(animationName);
+  animationNameToIndex.set(animationName, 0);
+
+  for (let i = 1; i < animationNameList.length; ++i) {
+    const nextAnimationName = animationNameList[i];
+    const nextAnimationIndex = animationNameToIndex.get(nextAnimationName);
+
+    if (nextAnimationIndex === undefined) {
+      throw new Error('[Reanimated] Failed to obtain animation index.');
+    }
+
+    animationNameToIndex.set(animationNameList[i], nextAnimationIndex + 1);
+  }
 }
 
 function removeWebAnimation(animationName: string) {
-  if (!customAnimations.has(animationName)) {
+  if (!isWindowAvailable()) {
```

Comment:
Can you add a comment here explaining why this check is necessary? @m-bert

Reply:
Sure - #5500